#	
#	File to drive the testing of the config file class
#	testGoal::config file::expected line Number of error (0) is success::expected error message
#
#	the loader expects a file with the same name as the config file ".py" to contain a
#	method called checkMethod.  This takes no parameters as config is global

Check Empty File::configfile_test1::config_file_not_called.py::0::Missing key -> 'LOG_DIRECTORY'

Not Accessible::configfile_test-not-there::config_file_not_called.py::0::Error accessing config file -> '[Errno 2] No such file or directory: '/Users/stu/Development/scripts/dtpo_autoload/tests/configFileTests/configfile_test-not-there''

Innaccessible Source Directory::noaccess/configfile_test3::config_file_not_called.py::0::Error accessing config file -> '[Errno 13] Permission denied: '/Users/stu/Development/scripts/dtpo_autoload/tests/configFileTests/noaccess/configfile_test3''

Missing Attributes::configfile_test4::config_file_not_called.py::37::Missing key -> 'ORPHAN_DOCUMENTS_DIRECTORY'

Duplicated Attributes::configfile_test5::config_file_not_called.py::43::Duplicate key -> 'PATTERN_FILE', now -> 'Duplicated Entry', was -> '/Users/stu/Development/scripts/dtpo_autoload/dtpo_autoload_patterns.conf''

Attributes with missing value (spaces)::configfile_test6::config_file_not_called.py::11::Missing value for key -> 'LOG_DIRECTORY'

Atributes with missing key::configfile_test7::config_file_not_called.py::39::Missing key -> 'LOG_DIRECTORY'

Good File with no Syntax variation::configfile_test8::check_good_config.py::0::???

Good file with offset comments etc::configfile_test9::check_good_config.py::0::???

Catch DEBUG enabled before Log Directory Set::configfile_test10::config_file_not_called.py::9::DEBUG enabled without LOG_DIRECTORY set

Good file with inaccessible log directory::configfile_test11::config_file_not_called.py::10::Log directory not accessible -> '/Users/stu/Development/scripts/dtpo_autoload/tests/configFileTests/noaccess', error -> directory not accessible. Error -> '[Errno 13] Permission denied: '/Users/stu/Development/scripts/dtpo_autoload/tests/configFileTests/noaccess/testFile.Remove''

Attributes with missing value (no spaces)::configfile_test12::config_file_not_called.py::39::Missing key -> 'LOG_DIRECTORY'

Good File but innaccessible working directory::configfile_test13::check_good_config_badsourcedir.py::38::SOURCE_DIRECTORY not accessible -> directory not accessible. Error -> '[Errno 13] Permission denied: '/Users/stu/Development/scripts/dtpo_autoload/tests/configFileTests/noaccess/testFile.Remove''

Pattern File not accessible::configfile_test14::check_good_config_badpatternfile.py::37::File PATTERN_FILE not accessible -> [Errno 13] Permission denied: '/Users/stu/Development/scripts/dtpo_autoload/tests/configFileTests/noaccess'
